tests/test_pytypes.cpp

```diff
@@ -923,4 +926,41 @@ TEST_SUBMODULE(pytypes, m) {
 #else
     m.attr("defined_PYBIND11_TYPING_H_HAS_STRING_LITERAL") = false;
 #endif
+
+#if defined(PYBIND11_HAS_RANGES) // test_ranges
+    m.attr("defined_PYBIND11_HAS_RANGES") = true;
+    m.def("iterator_default_initialization", []() {
+        using TupleIterator = decltype(py::tuple{}.begin());
+        using ListIterator = decltype(py::list{}.begin());
+        using DictIterator = decltype(py::dict{}.begin());
+        return py::bool_(
+            TupleIterator{} == TupleIterator{} && ListIterator{} == ListIterator{}
```

Comment:
Could you please make this more granular?

I'd just `m.def()`for `Tuple`, `List`, `Dict` separately.

If you lump them together, if something goes wrong (refactoring/cleanup), it's not obvious what's broken (one? or all? or some?).